### PR TITLE
Fix #5263: Site settings post formats now use FluxC

### LINKED ISSUE #5263: FluxC: Remove the last bits of XMLRPC code in wpandroid
Replace [this XMLRPC call](https://github.com/wordpress-mobile/WordPress-Android/blob/2528dba2cf79aa92a4ee6bd6f6b0862a4ce952c4/WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java#L775-L821) by broadcasting a `FETCH_POST_FORMATS`.

### PR DESCRIPTION
**Pull request by maxme**
Fix #5263: Site settings post formats now use FluxC.

This also fix an issue with Post Formats in Post Settings.

And I "accidentally" removed the directory `WordPress/src/main/java/org/xmlrpc/` 😁

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 1]
```diff
@@ -263,46 +263,35 @@ public void onClick(View view) {
...
-                    mPostFormatTitles[i] = getResources().getString(R.string.post_format_standard);
-                    i += 1;
-                }
-                for (PostFormatModel postFormatModel : postFormatModels) {
-                    mPostFormats[i] = postFormatModel.getSlug();
-                    mPostFormatTitles[i] = postFormatModel.getDisplayName();
-                    i += 1;
-                }
+            for (PostFormatModel postFormatModel : postFormatModels) {
+                mPostFormatKeys.add(postFormatModel.getSlug());
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: Here we're not replacing existing values, so we end up with many duplicate post formats.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 2]
```diff
@@ -263,46 +263,35 @@ public void onClick(View view) {
...
+                mPostFormatNames.add(postFormatModel.getDisplayName());
+            }
+            // Prepend the "Standard" post format if it's not already in
+            if (mPostFormatKeys.contains(POST_FORMAT_STANDARD_KEY)) {
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: I think there's a logic error here, should be `!mPostFormatKeys.contains(...`,

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: Fixed in  	c780e1b

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java (hunk 2 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: I'm not sure why this code existed in the first place, the "standard" post format was already in the list, I just moved it to the top.

**Comment by aforcier** [comment]
This looks good, I found one small issue that should be addressed in FluxC, opened a new issue: https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/333.

I'm not sure we should always show the default set of post formats no matter what the API gives us, but since it seems necessary for WP.com we may as well be consistent for now. I added a note about specifying `show-supported` for XML-RPC post formats [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/333#issuecomment-282486050), if we implement that we might consider changing the behavior in WPAndroid and only showing formats returned from the server, for XML-RPC at least.

Not a blocker for this PR though.

File: WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java [hunk 3]
```diff
@@ -1,15 +0,0 @@
-package org.xmlrpc.android;
-
-public class ApiHelper {
```
Comment (aforcier) [inline WordPress/src/main/java/org/xmlrpc/android/ApiHelper.java]: 💀 bye `ApiHelper`, you did your best.

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**